### PR TITLE
Adding fixes for the docker0 ipv6 issue.

### PR DESCRIPTION
**Pull request by rraghav-cisco**
### Description of PR

Summary:
In qos-sai-base, there is a docker0 ipv6 checking function, which fails if the DUT has no ipv6 address. The failure signature is as below:
```
14:50:01 __init__._fixture_generator_decorator    L0088 ERROR  | 
IndexError('list index out of range')
Traceback (most recent call last):
  File "/data/tests/common/plugins/log_section_start/__init__.py", line 84, in _fixture_generator_decorator
    res = next(it)
  File "/data/tests/qos/qos_sai_base.py", line 1824, in dut_disable_ipv6
    duthost.shell("sudo ip -6  addr show dev docker0 | grep global" + " | awk '{print $2}'")[
IndexError: list index out of range
```

### Type of change

- [X] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [X] 202405

#### What is the motivation for this PR?
The failure of the ipv6 disabling fixture.

#### How did you do it?
Check if the docker0 has ipv6 address or not.

#### How did you verify/test it?
Ran it on my TB:
```
----------------------------------------------------------------------------- live log sessionfinish -----------------------------------------------------------------------------
07:42:47 __init__.pytest_terminal_summary         L0067 INFO   | Can not get Allure report URL. Please check logs
============================================================================ short test summary info =============================================================================
PASSED qos/test_qos_sai.py::TestQosSai::testParameter[single_asic]
PASSED qos/test_qos_sai.py::TestQosSai::testParameter[single_dut_multi_asic]
PASSED qos/test_qos_sai.py::TestQosSai::testParameter[multi_dut_longlink_to_shortlink]
PASSED qos/test_qos_sai.py::TestQosSai::testParameter[multi_dut_shortlink_to_shortlink]
PASSED qos/test_qos_sai.py::TestQosSai::testParameter[multi_dut_shortlink_to_longlink]
=================================================================== 5 passed, 1 warning in 5246.68s (1:27:26) ====================================================================
sonic@202405-qos-sonic-mgmt-prod:/data/tests$ 
```

### CODE REVIEW

**Comment by wsycqyz** [comment]
@XuChen-MSFT for VIS

File: tests/qos/qos_sai_base.py [hunk 1]
```diff
@@ -1840,19 +1840,25 @@ def populateArpEntries(
...
         for duthost in get_src_dst_asic_and_duts['all_duts']:
-            docker0_ipv6_addr = \
-                duthost.shell("sudo ip -6  addr show dev docker0 | grep global" + " | awk '{print $2}'")[
-                    "stdout_lines"][0]
+            try:
+                all_docker0_ipv6_addrs[duthost.hostname] = \
+                    duthost.shell("sudo ip -6  addr show dev docker0 | grep global" + " | awk '{print $2}'")[
+                        "stdout_lines"][0]
+            except IndexError:
+                all_docker0_ipv6_addrs[duthost.hostname] = None
```
Comment (sdszhang) [inline tests/qos/qos_sai_base.py:1850]: can you add a log message for this exception?

Comment (rraghav-cisco) [reply tests/qos/qos_sai_base.py:1853]: > can you add a log message for this exception?
@sdszhang : Done.

File: tests/qos/qos_sai_base.py [hunk 2]
```diff
@@ -1840,19 +1840,25 @@ def populateArpEntries(
...
 
         for duthost in get_src_dst_asic_and_duts['all_duts']:
             duthost.shell("sysctl -w net.ipv6.conf.all.disable_ipv6=0")
-            logger.info("Adding docker0's IPv6 address since it was removed when disabing IPv6")
-            duthost.shell("ip -6 addr add {} dev docker0".format(docker0_ipv6_addr))
-            config_reload(duthost, config_source='config_db', safe_reload=True, check_intf_up_ports=True)
+            if all_docker0_ipv6_addrs[duthost.hostname] is not None:
+                logger.info("Adding docker0's IPv6 address since it was removed when disabing IPv6")
+                duthost.shell("ip -6 addr add {} dev docker0".format(all_docker0_ipv6_addrs[duthost.hostname]))
+                config_reload(duthost, config_source='config_db', safe_reload=True, check_intf_up_ports=True)
```
[the remark this answers is not in the document]
Comment (rraghav-cisco) [reply tests/qos/qos_sai_base.py:1864]: @sdszhang , good catch. Yes, I will implement this.

[the remark this answers is not in the document]
Comment (sdszhang) [reply tests/qos/qos_sai_base.py:1864]: @rraghav-cisco can you update this?

**Comment by sdszhang** [comment]
@rraghav-cisco can you update this PR? Thanks.

File: tests/qos/qos_sai_base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rraghav-cisco) [reply tests/qos/qos_sai_base.py]: @sdszhang , implemented the change, moved the config reload to a seperate block. Run result:
`=========================================================================================================================== PASSES ===========================================================================================================================
___________________________________________________________________________________________________________ TestQosSai.testParameter[single_asic] ____________________________________________________________________________________________________________
______________________________________________________________________________________________________ TestQosSai.testParameter[single_dut_multi_asic] _______________________________________________________________________________________________________
____________________________________________________________________________________________________________ TestQosSai.testParameter[multi_dut] _____________________________________________________________________________________________________________
________________________________________________________________________________________________________ TestQosSai.testParameter[multi_dut_reverse] _________________________________________________________________________________________________________
_________________________________________________________________________________________________ TestQosSai.testParameter[multi_dut_longlink_to_shortlink] __________________________________________________________________________________________________
_________________________________________________________________________________________________ TestQosSai.testParameter[multi_dut_shortlink_to_shortlink] _________________________________________________________________________________________________
_________________________________________________________________________________________________ TestQosSai.testParameter[multi_dut_shortlink_to_longlink] __________________________________________________________________________________________________
------------------------------------------------------ generated xml file: /run_logs/17467/2024-10-29-05-14-24/AJ-100g-SS-gb-pac/qos/test_qos_sai.py::TestQosSai::testParameter_2024-10-30-18-27-19.xml ------------------------------------------------------
INFO:root:Can not get Allure report URL. Please check logs
------------------------------------------------------------------------------------------------------------------- live log sessionfinish -------------------------------------------------------------------------------------------------------------------
20:25:41 __init__.pytest_terminal_summary         L0067 INFO   | Can not get Allure report URL. Please check logs
================================================================================================================== short test summary info ===================================================================================================================
PASSED qos/test_qos_sai.py::TestQosSai::testParameter[single_asic]
PASSED qos/test_qos_sai.py::TestQosSai::testParameter[single_dut_multi_asic]
PASSED qos/test_qos_sai.py::TestQosSai::testParameter[multi_dut]
PASSED qos/test_qos_sai.py::TestQosSai::testParameter[multi_dut_reverse]
PASSED qos/test_qos_sai.py::TestQosSai::testParameter[multi_dut_longlink_to_shortlink]
PASSED qos/test_qos_sai.py::TestQosSai::testParameter[multi_dut_shortlink_to_shortlink]
PASSED qos/test_qos_sai.py::TestQosSai::testParameter[multi_dut_shortlink_to_longlink]
========================================================================================================= 7 passed, 1 warning in 7100.31s (1:58:20) ==========================================================================================================
sonic@202405-qos-sonic-mgmt-prod:/data/tests$ 
`

**Pull request merged**